pkg/middleware/auth.go

```diff
@@ -0,0 +1,65 @@
+package middleware
+
+import (
+	"net/http"
+
+	clouddriver "github.com/billiford/go-clouddriver/pkg"
+	"github.com/billiford/go-clouddriver/pkg/fiat"
+	"github.com/gin-gonic/gin"
+)
+
+//authApplication takes a list of permissions
+//authAccount
+//go:generate counterfeiter . Auth
+type Auth interface {
+	AuthApplication() gin.HandlerFunc
+}
+
+type permissions struct {
+	list []string
+}
+
+func NewPermissions(list []string) permissions {
+	return permissions{list: list}
+}
+
+func (p *permissions) AuthApplication() gin.HandlerFunc {
```

Comment:
We don't need a pointer receiver for this func, just make it `func AuthApplication(permissions []string) gin.HandlerFunc`.